src/libraries/System.IO.FileSystem/tests/FileStream/CharacterDevice.cs

```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    [PlatformSpecific(TestPlatforms.AnyUnix)]
+    public class CharacterDevice
+    {
+        const string CharacterDevicePath = "/dev/tty";
+        [Theory]
+        [InlineData(FileOptions.None)]
+        [InlineData(FileOptions.Asynchronous)]
+        public void CharacterDevice_Write(FileOptions fileOptions)
+        {
+            using FileStream fs = new(CharacterDevicePath, new FileStreamOptions { Options = fileOptions, Access = FileAccess.Write });
+            fs.Write(Encoding.UTF8.GetBytes("foo"));
+        }
+
+        [Theory]
+        [InlineData(FileOptions.None)]
+        [InlineData(FileOptions.Asynchronous)]
+        public async Task CharacterDevice_WriteAsync(FileOptions fileOptions)
+        {
+            using FileStream fs = new(CharacterDevicePath, new FileStreamOptions { Options = fileOptions, Access = FileAccess.Write  });
+            await fs.WriteAsync(Encoding.UTF8.GetBytes("foo"));
+        }
```

Comment:
We were also getting the same ENXIO error on Read. I was trying to figure out how to test that with `/dev/tty` but I couldn't. Any suggestions are appreciated.

Reply:
For read, I think we would need to wait for https://github.com/dotnet/runtime/pull/58434 to see how it turns out. That PR may require some handling on Unix (as currently the code changes seem to be Windows specific, test is failing with possible ENXIO on write and read comes after).

Reply:
> I think we would need to wait for #58434 to see how it turns out.

That PR will not be backported to .NET 6.

A fix for this issue should be.

Reply:
Yup. The regression was found in write case only (as reading from unseekable file is not supported until now).